src/Microsoft.AspNetCore.Hosting/WebHostBuilderExtensions.cs

```diff
@@ -29,15 +29,21 @@ public static IWebHostBuilder Configure(this IWebHostBuilder hostBuilder, Action
 
             var startupAssemblyName = configureApp.GetMethodInfo().DeclaringType.GetTypeInfo().Assembly.GetName().Name;
 
-            return hostBuilder
-                .UseSetting(WebHostDefaults.ApplicationKey, startupAssemblyName)
-                .ConfigureServices(services =>
+            hostBuilder.UseSetting(WebHostDefaults.ApplicationKey, startupAssemblyName);
+
+            // Light up the GenericWebHostBuilder implementation
+            if (hostBuilder is GenericWebHostBuilder genericWebHostBuilder)
```

Comment:
Rather than sniffing for a specific type can we at least sniff for a new interface like IConfigureStartup with Configure, UseStartup, etc..

Reply:
Sure but why?

Reply:
Nah. It’s overkill and can be added at anytime. I would do that if we had something else concrete that would implement. Also I plan to keep this all internal as we’re not trying to expose anything public.

So in the end we’d have a new internal interface implemented and an internal type that would allow us to test these extension methods. I don’t see the value.

Reply:
👃

Reply:
Added them where?

An interface per extension seems overkill as they're all related to Startup/Configure.